File: C#.Net/JDI/Web/JDI Web/Selenium/Elements/Common/Button.cs
using System;
using Epam.JDI.Core.Interfaces.Common;
using JDI_Web.Selenium.Base;
using JDI_Web.Selenium.Elements.Base;
using OpenQA.Selenium;

namespace JDI_Web.Selenium.Elements.Common
{
    public class Button : ClickableText, IButton
    {
        public Button() : this(null) { }
        public Button(By byLocator = null, IWebElement webElement = null, WebBaseElement element = null)
            : base(byLocator, webElement, element:element) { }

        protected new Func<WebBaseElement, string> GetTextFunc =
           el => el.WebAvatar.FindImmediately(() => el.WebElement.GetAttribute("value"), "");
    }
}

File: Java/JDI/jdi-uitest-win/src/main/java/com/epam/jdi/uitests/win/winnium/elements/common/Button.java
package com.epam.jdi.uitests.win.winnium.elements.common;

import com.epam.jdi.uitests.core.interfaces.common.IButton;
import com.epam.jdi.uitests.win.winnium.elements.base.Element;
import com.epam.jdi.uitests.win.winnium.elements.base.managers.AttributeTextManager;
import com.epam.jdi.uitests.win.winnium.elements.base.managers.ClickableManager;
import org.openqa.selenium.By;

public class Button extends Element implements IButton {
    private ClickableManager clickableManager = new ClickableManager(this);
    private AttributeTextManager textFromAttribute = new AttributeTextManager(this);

    public Button() {}
    public Button(By locator) { setLocator(locator);}

    @Override
    public String getText() {
        return textFromAttribute.getText();
    }

    @Override
    public String getValue() {
        return textFromAttribute.getValue();
    }

    @Override
    public String waitText(String text) {
        return textFromAttribute.waitText(text);
    }

    @Override
    public String waitMatchText(String regEx) {
        return textFromAttribute.waitMatchText(regEx);
    }

    @Override
    public void click() {
        clickableManager.click();
    }
}
